File: b_rank_new_level_up_problems/b_rank_new_level_up_problems__get_row_col.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;

class Program
{
    static void Main()
    {
        int[] hw = ReadLine().Split( ' ' ).Select( int.Parse ).ToArray();
        bool[,] b = new bool[hw[0], hw[1]];
        var sb = new System.Text.StringBuilder();

        foreach( int i in Range( 0, hw[0] ) ) {
            string s = ReadLine();

            foreach( int j in Range( 0, hw[1] ) ) {
                b[i, j] = s[j] == '#';
            }
        }

        int[] yx = ReadLine().Split( ' ' ).Select( int.Parse ).ToArray();

        foreach( int i in Range( 0, hw[0] ) ) {
            foreach( int j in Range( 0, hw[1] ) ) {
                if( i + j == yx[0] + yx[1] | i - j == yx[0] - yx[1] | i == yx[0] | j == yx[1] ) {
                    b[i, j] = !b[i, j];
                }
            }
        }
        foreach( int i in Range( 0, hw[0] ) ) {
            foreach( int j in Range( 0, hw[1] ) ) {
                sb.Append( b[i, j] ? '#' : '.' );
            }
            sb.AppendLine();
        }
        WriteLine( sb );
    }
}

File: b_rank_new_level_up_problems/b_rank_new_level_up_problems__get_row_col.py
# 実行時間 10ms
import sys

rl = sys.stdin.readline
h, w = list(map(int, rl().split()))
b = [[False for _ in [0] * w] for _ in [0] * h]

for i in range(h):
    s = rl()

    for j in range(w):
        b[i][j] = s[j] == "#"

y, x = list(map(int, rl().split()))

for i in range(h):
    for j in range(w):
        if i + j == y + x or i - j == y - x or i == y or j == x:
            b[i][j] = not b[i][j]
for i in range(h):
    for j in range(w):
        print("#" if b[i][j] else ".", end="")
    print()
